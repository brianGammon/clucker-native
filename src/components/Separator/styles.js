import { StyleSheet } from 'react-native';
import variables from '../../styles/variables';

export default StyleSheet.create({
  separator: {
    backgroundColor: variables.cluckerOrange,
  },
  separatorLabel: {
    color: variables.cluckerGrey,
    fontWeight: 'bold',
  },
});
