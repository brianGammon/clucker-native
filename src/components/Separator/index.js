import React from 'react';
import { Text, Separator as NBSeparator } from 'native-base';
import styles from './styles';

type Props = {
  text: string,
};

const Separator = ({ text }: Props) => (
  <NBSeparator style={styles.separator}>
    <Text style={styles.separatorLabel}>{text}</Text>
  </NBSeparator>
);

export default Separator;
