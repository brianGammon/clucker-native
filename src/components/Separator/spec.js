import React from 'react';
import Separator from '.';

describe('Separator', () => {
  test('renders with text', () => {
    const wrapper = shallow(<Separator text="Test Message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
