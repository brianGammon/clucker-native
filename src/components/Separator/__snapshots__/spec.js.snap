// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders with text 1`] = `
<Styled(Separator)
  style={
    Object {
      "backgroundColor": "rgb(249, 149, 0)",
    }
  }
>
  <Styled(Text)
    style={
      Object {
        "color": "#575757",
        "fontWeight": "bold",
      }
    }
  >
    Test Message
  </Styled(Text)>
</Styled(Separator)>
`;
